### PR TITLE
Update tracking conditions

### PR DESCRIPTION
**Pull request by malithsen**
#### Changes proposed in this Pull Request

This PR:

- Disables tracking when WooPayments gateway is disabled
- Disables Button location updates when WooPay is disabled

#### Test disabled gateway
1. Add a breakpoint [here](https://github.com/Automattic/woocommerce-payments/blob/6ea513ef9ec64812aa214af8779b376499b74b22/includes/class-woopay-tracker.php#L249) 
2. As a non-admin customer, visit the checkout page. Make sure the breakpoint in step 1 is reached
3. Disable WooPayments from WooCommerce -> Payments
4. As a non-admin customer, visit the checkout page. Make sure the breakpoint in step 1 is not reached.

#### Test button updates event
1. With WooPay enabled, update the options in "Enable WooPay button on selected pages" section 
2. click "Save Changes" 
3. In ~5 minutes check MC Tracks Live view for the `wcadmin_woopay_express_button_locations_updated` event
4. Disable WooPay 
5. Click "Save Changes" in Payments Settings.
6. There should not be an event for `wcadmin_woopay_express_button_locations_updated` from your blog.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7244` or branch name `fix/tracking-conditions` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/tracking-conditions) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 72ca17dd9d9bbd1444baa432092be9225f962e81
- Build time: 2023-10-02 18:23:02 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/admin/class-wc-rest-payments-settings-controller.php [hunk 1]
```diff
@@ -881,6 +881,11 @@ private function update_woopay_enabled_locations( WP_REST_Request $request ) {
 			return;
 		}
 
+		$is_woopay_enabled = 'yes' === $this->wcpay_gateway->get_option( 'platform_checkout', 'no' );
```
Comment (hsingyuc) [inline includes/admin/class-wc-rest-payments-settings-controller.php]: Can we reuse the [`is_woopay_enabled` function](https://github.com/Automattic/woocommerce-payments/blob/6ea513ef9ec64812aa214af8779b376499b74b22/includes/class-wc-payments-features.php#L125) here?

**Review by hsingyuc — COMMENTED** [review]
@malithsen Thank you for working on this!  Tests are as expected and I left one question below.

File: includes/admin/class-wc-rest-payments-settings-controller.php (hunk 1 above)
Comment (malithsen) [reply includes/admin/class-wc-rest-payments-settings-controller.php]: ah thanks! I forgot about that function. Updated in https://github.com/Automattic/woocommerce-payments/pull/7244/commits/7250cb687f242797fd93e223877e749bff6c5fe7

**Review by hsingyuc — APPROVED** [review]
Thank you for the fixes @malithsen! LGTM!

**Pull request merged**